src/features/hyperlinks.rs

```diff
@@ -19,26 +19,83 @@ pub fn make_feature() -> Vec<(String, OptionValueFunction)> {
     ])
 }
 
+lazy_static! {
+    static ref COMMIT_LINE_REGEX: Regex = Regex::new(r"\b[0-9a-f]{8,40}\b").unwrap();
+}
+
+#[cfg(test)]
+pub fn remote_from_config(_: &Option<&GitConfig>) -> Option<GitRemoteRepo> {
+    Some(GitRemoteRepo::GitHub {
+        slug: "dandavison/delta".to_string(),
+    })
+}
+
+#[cfg(not(test))]
+pub fn remote_from_config(cfg: &Option<&GitConfig>) -> Option<GitRemoteRepo> {
+    cfg.and_then(GitConfig::get_remote_url)
+}
+
 pub fn format_commit_line_with_osc8_commit_hyperlink<'a>(
     line: &'a str,
     config: &Config,
 ) -> Cow<'a, str> {
+    // Given a [START..END]-match and pos = 0, store [pos..START] first, then the hyperlink(match),
+    // then store pos = END. Repeat while matches exist, and at the end store [pos..].
     if let Some(commit_link_format) = &config.hyperlinks_commit_link_format {
-        COMMIT_LINE_REGEX.replace(line, |captures: &Captures| {
-            let prefix = captures.get(1).map(|m| m.as_str()).unwrap_or("");
-            let commit = captures.get(2).map(|m| m.as_str()).unwrap();
-            let suffix = captures.get(3).map(|m| m.as_str()).unwrap_or("");
-            let formatted_commit =
-                format_osc8_hyperlink(&commit_link_format.replace("{commit}", commit), commit);
-            format!("{prefix}{formatted_commit}{suffix}")
-        })
-    } else if let Some(repo) = config.git_config().and_then(GitConfig::get_remote_url) {
-        COMMIT_LINE_REGEX.replace(line, |captures: &Captures| {
-            format_commit_line_captures_with_osc8_commit_hyperlink(captures, &repo)
-        })
-    } else {
-        Cow::from(line)
+        let mut matches = COMMIT_LINE_REGEX.find_iter(line);
+        if let Some(capture) = matches.next() {
+            let mut pos = 0;
+            let mut result = String::new();
+
+            let mut create_commit_hyperlink = |pre_cap_pos, cap: Match| {
+                result.push_str(&line[pre_cap_pos..cap.start()]);
+                let commit = &line[cap.start()..cap.end()];
+                let formatted_commit =
+                    format_osc8_hyperlink(&commit_link_format.replace("{commit}", commit), commit);
```

Comment:
What about filtering out pure numbers here again, as mentioned in the issue?

Reply:
Yes, why not. The posterior probability of such a string being a number is going to be considerably higher than that of it being a commit. Multiplied by the probability that someone happened to care about the link to their improbably-digit-rich commit...

Reply:
Done.